### PR TITLE
APPS-1598: document building dx-toolkit in docker

### PR DESCRIPTION
**Pull request by piader1**
Document how to build dx-toolkit from inside docker.
https://jira.internal.dnanexus.com/browse/APPS-1598

### CODE REVIEW

**Comment by piader1** [comment]
Hi, added a guide on how to build dx-toolkit inside docker. I know some of you are using this so let me know what you think.

File: Readme-developer.md [hunk 1]
```diff
@@ -47,6 +49,28 @@ https://documentation.dnanexus.com/downloads
     make
     ```
 
+# Building inside docker 
+
+To avoid lengthy installation of dependencies on osx and simultaneous installations of development versions of dx-toolkit, you can build your dev-version of `dx-toolkit` in the docker container.
```
Comment (jkotrs) [inline Readme-developer.md:54]: "OS X" to stay consistent?

File: Readme-developer.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Building on OS X 10.9 & 11.4
+Building from source
```
Comment (mhrvol) [inline Readme-developer.md:1]: I believe there is Readme.md that has information for not-osx operation systems., maybe we could move your additions to a separate readme? Or probably put it in that readme?

File: Readme-developer.md [hunk 3]
```diff
@@ -47,6 +49,28 @@ https://documentation.dnanexus.com/downloads
...
+    ```
+    # from root folder of dx-tollkit
+    docker run -v `pwd`:/dx-toolkit -w /dx-toolkit -it --rm --entrypoint=/bin/bash python:3.9-bullseye
+    ```
+2. From the interactive shell install `dx-toolkit`:
+
+    ```
+    python -m pip install src/python/ --upgrade
+    ```
+3. Log in and use the container while developing. To rebuild, just save the work and run the step 2 again.
```
Comment (mhrvol) [inline Readme-developer.md:67]: We could probably have 2a and 2b, as they are functionality the same, right?

File: Readme-developer.md [hunk 4]
```diff
@@ -47,6 +49,28 @@ https://documentation.dnanexus.com/downloads
...
+    docker run -v `pwd`:/dx-toolkit -w /dx-toolkit -it --rm --entrypoint=/bin/bash python:3.9-bullseye
+    ```
+2. From the interactive shell install `dx-toolkit`:
+
+    ```
+    python -m pip install src/python/ --upgrade
+    ```
+3. Log in and use the container while developing. To rebuild, just save the work and run the step 2 again. 
+
+Also, you can isntall the the `dx-toolkit` from remote like so:
```
Comment (mhrvol) [inline Readme-developer.md:69]:
```suggestion
Also, you can install the `dx-toolkit` from remote like so: 
```

File: Readme-developer.md [hunk 5]
```diff
@@ -47,6 +49,28 @@ https://documentation.dnanexus.com/downloads
...
+2. From the interactive shell install `dx-toolkit`:
+
+    ```
+    python -m pip install src/python/ --upgrade
+    ```
+3. Log in and use the container while developing. To rebuild, just save the work and run the step 2 again. 
+
+Also, you can isntall the the `dx-toolkit` from remote like so: 
+    ```
+    python3 -m pip install --upgrade 'git+https://github.com/dnanexus/dx-toolkit.git@master#egg=dxpy&subdirectory=src/python'
```
Comment (mhrvol) [inline Readme-developer.md:71]: This might be 2a (or 2b), so user immediately knows there are more options?

Comment (piader1) [reply Readme-developer.md:71]: Sure.

File: Readme-developer.md (hunk 2 above)
Comment (piader1) [reply Readme-developer.md:1]: I will have you to decide that - what is the best?

Comment (piader1) [reply Readme-developer.md:1]: @mhrvol

Comment (piader1) [reply Readme-developer.md:1]: I will add this to the readme in dx-toolkit/src/python/Readme.md - Building section

**Review by kpjensen — APPROVED** [review]

**Pull request merged**